src/Mailer/MailPreview.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.3
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace DebugKit\Mailer;
+
+use Cake\Mailer\MailerAwareTrait;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionMethod;
+
+class MailPreview
+{
+    use MailerAwareTrait;
+
+    /**
+     * Returns the name of an email if it is valid
+     *
+     * @param string $email Email name
+     * @return bool|string
+     **/
+    public function find($email)
+    {
+        if ($this->validEmail($email)) {
+            return $email;
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns a list of valid emails
+     *
+     * @return array
+     **/
+    public function getEmails()
+    {
+        $emails = [];
+        foreach (get_class_methods($this) as $methodName) {
+            if (!$this->validEmail($methodName)) {
+                continue;
+            }
+
+            $emails[] = $methodName;
+        }
+
+        return $emails;
+    }
+
+    /**
+     * Returns the name of this preview
+     *
+     * @return string
+     **/
+    public function name()
+    {
+        $classname = get_class($this);
+        if ($pos = strrpos($classname, '\\')) {
+            return substr($classname, $pos + 1);
+        }
+
+        return $pos;
```

Comment:
This looks like it could return mixed string|int ? Should be null instead if not found?